qa/fips/docker/Dockerfile

```diff
@@ -0,0 +1,72 @@
+# Start from the FIPS-compliant base image
+FROM docker.elastic.co/wolfi/chainguard-base-fips:latest
+
+# Install OpenJDK 21
+RUN apk add --no-cache \
+    openjdk-21 \
+    bash
+
+# Create directory for security configuration
+RUN mkdir -p /etc/java/security
+RUN mkdir -p /root/.gradle
+
+# Copy configuration files
+COPY qa/fips/config/security/java.security /etc/java/security/
+COPY qa/fips/config/security/java.policy /etc/java/security/
+
+# Set environment variables
+ENV JAVA_HOME=/usr/lib/jvm/java-21-openjdk
+ENV PATH="${JAVA_HOME}/bin:${PATH}"
+
+# Create working directory
+WORKDIR /logstash
+
+# Copy the local Logstash source
+COPY . .
+
+# Initial build using JKS truststore
+RUN ./gradlew clean bootstrap assemble installDefaultGems
+
+# Convert JKS to BCFKS for truststore and keystore
+RUN keytool -importkeystore \
+    -srckeystore $JAVA_HOME/lib/security/cacerts \
+    -destkeystore /etc/java/security/cacerts.bcfks \
+    -srcstoretype jks \
+    -deststoretype bcfks \
+    -providerpath /logstash/logstash-core/lib/jars/bc-fips-2.0.0.jar \
+    -provider org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider \
+    -deststorepass changeit \
+    -srcstorepass changeit \
+    -noprompt
+
+RUN keytool -importkeystore \
+    -srckeystore $JAVA_HOME/lib/security/cacerts \
+    -destkeystore /etc/java/security/keystore.bcfks \
+    -srcstoretype jks \
+    -deststoretype bcfks \
+    -providerpath /logstash/logstash-core/lib/jars/bc-fips-2.0.0.jar \
+    -provider org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider \
+    -deststorepass changeit \
+    -srcstorepass changeit \
+    -noprompt
+
+ENV JAVA_SECURITY_PROPERTIES=/etc/java/security/java.security
+ENV JAVA_OPTS="\
+    -Dio.netty.ssl.provider=JDK \
```

Comment:
We probably want this so that it does not try to use openssl as default https://netty.io/4.1/api/io/netty/handler/ssl/SslContext.html#defaultClientProvider--